src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+import {Position} from './Position'
+
+export class Matrix{
+    public board:number[][] = [];
+    /*
+    0 = uncharted
+    1 = hit
+    2 = miss
+    3 = suggested target
+    4 = blocked out no ship area
+    */
+    public ships:boolean[] = [true,true,true,true,true];
+    /*
+    index = length
+    0 = 2
+    1 = 3
+    2 = 3
+    3 = 4
+    4 = 5
+     */
```

Comment:
If you're putting in these comments to explain variables, you know that you've probably missed a trick.

In this case - why not create classes for them? Instead of having `number[][]` for board, why not set up a class to carry the actual names & definitions - other better yet, and [enum](https://www.typescriptlang.org/docs/handbook/enums.html)! (def. read that, it's a perfect fit!)

And with the ships, why not have a `ShipList` class with named variables for ship1, ship2, etc (or even better, `battleship`, `destroyer`, etc!)